hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/FromElement.java

```diff
@@ -279,7 +279,7 @@ protected void appendDisplayText(StringBuilder buf) {
 		if ( columns != null ) {
 			for ( int i = 0; i < columns.length; i++ ) {
 				buf.append( columns[i] );
-				if ( i < columns.length ) {
+				if ( i < columns.length - 1 ) {
```

Comment:
the original test always evaluates to true, so I'm assuming the intention was to omit the last element